pkg/controller/elasticsearch/driver/upgrade_pods_deletion.go

```diff
@@ -83,11 +83,13 @@ func (ctx *rollingUpgradeCtx) getAllowedDeletions() (int, bool) {
 	// Upscale is done, we should have the required number of Pods
 	expectedPods := ctx.statefulSets.PodNames()
 	unhealthyPods := len(expectedPods) - len(ctx.healthyPods)
-	maxUnavailable := 1
-	if ctx.ES.Spec.UpdateStrategy.ChangeBudget != nil {
-		maxUnavailable = ctx.ES.Spec.UpdateStrategy.ChangeBudget.MaxUnavailable
+
+	maxUnavailable := ctx.ES.Spec.UpdateStrategy.ChangeBudget.GetMaxUnavailableOrDefault()
+	if maxUnavailable == nil {
+		return len(expectedPods), false
```

Comment:
`// maxUnavailable is unbounded`

Reply:
Added. 
Also, I'll rename `expectedPods` to `actualPods` as we tend to use `actual` when talking about current state vs `expected` which is the future state we want to achieve. Here, the `expectedPods` represent the current state, hence the rename.